agent/__init__.py

```diff
@@ -639,61 +635,72 @@ def write_metadata(data, rewrite_file):
     metadata_path.chmod(0o644)
 
 
-def run(ping=True, debug=False, dev=False):
+def run(ping=True, dev=False, logger=logger, daemon=False):
+
+    def log_or_print(msg):
+        if daemon:
+            logger.info(msg)
+        else:
+            print(msg)
```

Comment:
Why did you leave this `print`? Python's logger can output to console.

Reply:
Just to let it on manual start to looks like that
```
2019-07-08 12:31:45,502 - agent - INFO - start in ping mode...
2019-07-08 12:31:45,516 - agent - WARNING - No certificate found on disk.
Got WoTT ID: 4285b1e1001644018ced2e52122ad0ea.d.wott.local
Generating certificate...
Submitting CSR...
Got Claim Token: 50786d63-7cc7-4143-9c1d-a8c2c29ce444
Claim your device: https://dash.wott.io/claim-device?device_id=4285b1e1001644018ced2e52122ad0ea.d.wott.local&claim_token=50786d63-7cc7-4143-9c1d-a8c2c29ce444
Writing certificate and key to disk...
Writing config...
```
but not like that

```
2019-07-08 12:34:43,093 - agent - INFO - start in ping mode...
2019-07-08 12:34:43,097 - agent - WARNING - No certificate found on disk.
2019-07-08 12:34:44,543 - agent - INFO - Got WoTT ID: 1975616a847745deb9eda038e84168d2.d.wott.local
2019-07-08 12:34:44,546 - agent - INFO - Generating certificate...
2019-07-08 12:34:45,757 - agent - INFO - Submitting CSR...
2019-07-08 12:34:47,128 - agent - INFO - Got Claim Token: d8d626f2-5e8e-4b53-a055-b86fff60ca14
2019-07-08 12:34:47,129 - agent - INFO - Claim your device: https://dash.wott.io/claim-device?device_id=1975616a847745deb9eda038e84168d2.d.wott.local&claim_token=d8d626f2-5e8e-4b53-a055-b86fff60ca14
2019-07-08 12:34:47,130 - agent - INFO - Writing certificate and key to disk...
2019-07-08 12:34:47,266 - agent - INFO - Writing config...
```
So if this is not needed, then i can leave just logger.info()

P.S. And if needed, then may be also change first two lines of output (or remove first and fix second)?

Reply:
changed